src/generators/typescript/TypeScriptRenderer.ts

```diff
@@ -209,8 +212,8 @@ ${lines.map(line => ` * ${line}`).join('\n')}
     return this.renderBlock(content);
   }
 
-  renderProperty(propertyName: string, property: CommonModel, type: PropertyType = PropertyType.property): string {
-    const formattedPropertyName = this.nameProperty(propertyName, property);
+  renderProperty(propertyName: string, property: CommonModel, type: PropertyType = PropertyType.property, checkReserved = true): string {
```

Comment:
I think that we should go with strict checking of the reserved keyword, even if TS allows it.

Reply:
That is also a possibility, it will make the implementation easier to understand as well, I am just not sure whether the end-user will find it weird 🤔

Reply:
I mean, in the JS you don't have such keywords like `enum`, `interface` etc, in TS you have - in some scope - so I opt to make it strict :) We can also add options to the TS generator if generator should be strict in this area, or not.

Reply:
such an option can be added in another PR.

Reply:
If the use-case ever comes up, it can of course be added then. Gonna change to a strict no keyword usage across both presets.